src/averageAges.js

```diff
@@ -55,7 +77,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenList = people.filter(child => {
+    return people.some(mother => mother.name === child.mother
+      && (onlyWithSon
+        ? child.sex === 'm'
+        : true)
+    );
+  });
+
+  const ageDifferences = childrenList.map(child => {
+    const foundMother = people.find(mother => mother.name === child.mother);
+
+    return child.born - foundMother.born;
+  });
+
+  return getAverageAge(ageDifferences, childrenList.length);
+}
+
+function getAverageAge(ages, divider) {
```

Comment:
You can use `ages.length` here I think, so you can omit second argument

Reply:
@dinaminahi thanks, you're right, here a lot of details, easy to miss them